From fbea1b4146fda7c7e5625dd6160d735a6c0ffe9c Mon Sep 17 00:00:00 2001
From: jagaur <jagaur@habana.ai>
Date: Tue, 03 Jun 2025 15:38:22 +0300
Subject: [PATCH] [SW-230770] gaudi3: print all crc sizes

test patch to print all crc32 sizes to isolate issue.

Change-Id: I8aebabd1c349d11217be926fbaf7656202f9fd5b
---

diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index d5726b9..bf10a81 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -130,9 +130,14 @@
 	FILE=$1
 	OUT_FILE=$2
 	WORD_SIZE=${3}
+
+	printf "[hl_reverse_words] Printing file, file size, word size, out file :"
 	local filesize=$(wc -c < $FILE)
-	echo $FILE $filesize $WORD_SIZE
+	echo $(basename $FILE) $filesize $WORD_SIZE $(basename $OUT_FILE)
 	${OBJCOPY}  -Ibinary --reverse-bytes=${WORD_SIZE} ${FILE} ${OUT_FILE}
+	filesize=$(wc -c < $OUT_FILE)
+	printf "[hl_reverse_words] Printing out file, out file size :"
+	echo $(basename $OUT_FILE) $filesize
 }
 
 function hl_calc_crc32 {
@@ -141,7 +146,12 @@
 	# reverse CRC32 source as DMA engine works like that
 	hl_reverse_words ${FILE} ${FILE}.out 4
 	CRC32=$(crc32 ${FILE}.out)
+	echo "CRC32: ${CRC32}"
 	echo ${CRC32} | xxd -r -p > ${FILE}.crc32
+	hexdump -C ${FILE}.crc32
+	printf "[hl_calc_crc32] Printing out file, out file size :"
+	local filesize=$(wc -c < ${FILE}.crc32)
+	echo $(basename ${FILE}.crc32) $filesize
 }
 
 function hl_img_offset {
diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index 6347012..f914782 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -290,13 +290,30 @@
 }
 
 function prep_img_desc_with_crc {
+	printf "\nExecuting function [prep_img_desc_with_crc]\n"
 	BINFILE=$1
 	OUTFILENAME=$(basename $1)
 	cat ${BINFILE}>${BINFILE}.tmp.pad
 	dd status=none conv=sync bs=16 if=${BINFILE}.tmp.pad of=${BINFILE}.tmp.algn
+
+	local filesize=$(wc -c < ${BINFILE}.tmp.algn)
+	printf "File 1, Size :"
+	echo $(basename ${BINFILE}.tmp.algn) $filesize
+
 	hl_calc_crc32 ${BINFILE}.tmp.algn
+
+	printf "File 2, Size :"
+	filesize=$(wc -c < ${BINFILE}.tmp.algn.crc32)
+	echo $(basename ${BINFILE}.tmp.algn.crc32) $filesize
+
 	cat ${BINFILE}.tmp.algn ${BINFILE}.tmp.algn.crc32 > ${BINS_DIR}/${DEV_PREFIX}-${OUTFILENAME}
-	rm -rf ${BINFILE}.tmp*
+
+	printf "Concat file, size:"
+	filesize=$(wc -c < ${BINS_DIR}/${DEV_PREFIX}-${OUTFILENAME})
+	echo $(basename ${BINS_DIR}/${DEV_PREFIX}-${OUTFILENAME}) $filesize
+	echo
+
+	#rm -rf ${BINFILE}.tmp*
 }
 
 function hl_chksum_add {
